checker/src/main/java/org/checkerframework/checker/optional/OptionalChecker.java

```diff
@@ -1,32 +1,37 @@
 package org.checkerframework.checker.optional;
 
-import java.util.Optional;
-import java.util.Set;
-import org.checkerframework.common.aliasing.AliasingChecker;
-import org.checkerframework.common.basetype.BaseTypeChecker;
-import org.checkerframework.framework.qual.RelevantJavaTypes;
-import org.checkerframework.framework.qual.StubFiles;
+import java.util.ArrayList;
+import java.util.Collection;
+import org.checkerframework.checker.nonempty.NonEmptyChecker;
+import org.checkerframework.framework.source.AggregateChecker;
 import org.checkerframework.framework.source.SourceChecker;
-import org.checkerframework.framework.source.SupportedOptions;
 
 /**
- * A type-checker that prevents misuse of the {@link java.util.Optional} class.
+ * A type-checker that prevents {@link java.util.NoSuchElementException} in the use of the {@link
+ * java.util.Optional} class.
+ *
+ * <p>The Optional Checker aggregates two checkers, the Non-Empty Checker and the
+ * OptionalImplChecker. The Optional Checker behaves as the ultimate parent of those two checkers
+ * and has no qualifiers or transfer function class of its own.
+ *
+ * <p>When the Non-Empty Checker is executed as a subchecker of the Optional Checker, it runs the
+ * OptionalImplChecker. It only type-checks the methods that the OptionalImplChecker passes to it.
+ *
+ * <p>The OptionalImplChecker implements a type system for Optional values (i.e., what users would
+ * expect the Optional Checker to implement). It reads programmer-written annotations from the
+ * Non-Empty type system, and keeps track of the methods that the Non-Empty Checker should check.
```

Comment:
@smillst I have written a summary of the interactions between the `OptionalChecker`, `NonEmptyChecker`, and the `OptionalImplChecker` here

Reply:
This great thanks!  It should just be a `\\` comment since it contains implementation details.